metatensor-torch/tests/cmake-project/CMakeLists.txt

```diff
@@ -2,14 +2,27 @@ cmake_minimum_required(VERSION 3.16)
 
 project(metatensor-torch-test-cmake-project CXX)
 
-# If building a dev version, we also need to update the REQUIRED_METATENSOR_VERSION
-# in the same way we update the metatensor-torch version
-include(../../cmake/dev-versions.cmake)
-set(REQUIRED_METATENSOR_TORCH_VERSION "0.1.0")
-create_development_version("${REQUIRED_METATENSOR_TORCH_VERSION}" "metatensor-torch-v" METATENSOR_TORCH_FULL_VERSION)
-string(REGEX REPLACE "([0-9]*)\\.([0-9]*).*" "\\1.\\2" REQUIRED_METATENSOR_TORCH_VERSION ${METATENSOR_TORCH_FULL_VERSION})
-
-find_package(metatensor_torch ${REQUIRED_METATENSOR_TORCH_VERSION} CONFIG REQUIRED)
+option(USE_CMAKE_SUBDIRECTORY OFF)
+
+if (USE_CMAKE_SUBDIRECTORY)
+    set(BUILD_METATENSOR_TORCH ON)
+    # build metatensor_torch as part of this project
+    add_subdirectory(../../../ metatensor)
+    find_package(Torch)
+
+    # load metatensor and metatensor-torch from the build path
+    set(CMAKE_BUILD_RPATH "$<TARGET_FILE_DIR:torch>;$<TARGET_FILE_DIR:metatensor::shared>;$<TARGET_FILE_DIR:metatensor_torch>")
+else()
+    # If building a dev version, we also need to update the REQUIRED_METATENSOR_VERSION
+    # in the same way we update the metatensor-torch version
+    include(../../cmake/dev-versions.cmake)
+    set(REQUIRED_METATENSOR_TORCH_VERSION "0.1.0")
```

Comment:
Couldn't we put a variable here. To avoid changing the version in many places when we bump the version?

Reply:
I don't think so. This is the same as pip dependencies declaration around here: https://github.com/lab-cosmo/metatensor/blob/b5480cbd8f5c2fc4f59ea812e8c7cabd64caad5f/python/metatensor-operations/setup.py#L129, which we also need to update for every release.

Reply:
okay